src/components/AnonymousChat/child/ChatItemMyTextV2.tsx

```diff
@@ -109,6 +116,50 @@ const ChatItemMyTextV2 = ({
                     <Text style={[styles.timeText, textStyle(true)]}>{time}</Text>
                   </View>
                 )}
+                {attachments.length > 0 && messageType !== 'deleted' && (
+                  <View style={styles.attachmentContainer}>
+                    {attachments
+                      .filter((item, index) => index <= 3)
+                      .map((item, index) => (
+                        <TouchableOpacity
+                          key={item.thumb_url}
+                          style={{
+                            width: `${
+                              (attachments.length >= 3 && index > 0) || attachments.length >= 4
+                                ? 50
+                                : 100
+                            }%`,
+                            height: `${attachments.length >= 3 ? 50 : 100 / attachments.length}%`,
+                            position: 'relative',
+                            overflow: 'hidden'
+                          }}
+                          activeOpacity={1}
+                          onPress={
+                            attachments.length > 0
+                              ? () => onOpenMediaPreview(attachments, index, navigation)
+                              : null
+                          }>
+                          <FastImage
+                            style={styles.image}
+                            source={{
+                              uri: item.thumb_url
+                            }}
+                          />
+                          {attachments.length > 4 && index === 3 && (
+                            <View style={styles.moreOverlay}>
+                              <Text style={styles.moreText}>+{attachments.length - 4}</Text>
+                            </View>
+                          )}
+                          {/* Video Play Icon */}
+                          {item.video_path && (
+                            <View style={styles.moreOverlay}>
+                              <IconVideoPlay />
+                            </View>
+                          )}
+                        </TouchableOpacity>
+                      ))}
+                  </View>
+                )}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This large block of code is responsible for rendering the attachments in the chat message. It includes logic for displaying up to four attachments with thumbnails, showing a "+n" overlay for more than four attachments, and handling clicks on the attachments to open media previews or play videos. The logic seems sound, but it's quite complex and could potentially be broken down into smaller components or helper functions for better readability and maintainability.
```diff
+                 {attachments.length > 0 && messageType !== 'deleted' && (
+                   <View style={styles.attachmentContainer}>
+                     {attachments
+                       .filter((item, index) => index <= 3)
+                       .map((item, index) => (
+                         <TouchableOpacity
+                           key={item.thumb_url}
+                           style={{
+                             width: `${
+                               (attachments.length >= 3 && index > 0) || attachments.length >= 4
+                                 ? 50
+                                 : 100
+                             }%`,
+                             height: `${attachments.length >= 3 ? 50 : 100 / attachments.length}%`,
+                             position: 'relative',
+                             overflow: 'hidden'
+                           }}
+                           activeOpacity={1}
+                           onPress={
+                             attachments.length > 0
+                               ? () => onOpenMediaPreview(attachments, index, navigation)
+                               : null
+                           }>
+                           <FastImage
+                             style={styles.image}
+                             source={{
+                               uri: item.thumb_url
+                             }}
+                           />
+                           {attachments.length > 4 && index === 3 && (
+                             <View style={styles.moreOverlay}>
+                               <Text style={styles.moreText}>+{attachments.length - 4}</Text>
+                             </View>
+                           )}
+                           {/* Video Play Icon */}
+                           {item.video_path && (
+                             <View style={styles.moreOverlay}>
+                               <IconVideoPlay />
+                             </View>
+                           )}
+                         </TouchableOpacity>
+                       ))}
+                   </View>
+                 )}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->